src/services/findAllReferences.ts

```diff
@@ -784,8 +784,8 @@ namespace ts.FindAllReferences.Core {
             return;
         }
 
-        const fullStart = state.options.findInComments || container.jsDoc !== undefined || forEach(search.symbol.declarations, d => d.kind === ts.SyntaxKind.JSDocTypedefTag);
-        for (const position of getPossibleSymbolReferencePositions(sourceFile, search.text, container, fullStart)) {
+        // Need to search in the full start of the node in case there is a reference inside JSDoc.
+        for (const position of getPossibleSymbolReferencePositions(sourceFile, search.text, container, /*fullStart*/ true)) {

```

Comment:
Wouldn't this have been `true` if `container.jsDoc !== undefined`, or are we opening this up to all comments of the container because we need to find it on a node where we don't currently parse JSDoc comments in the parser?